src/entity/Ship.java

```diff
@@ -401,24 +397,43 @@ public void subtractHealth(){
 	//barrier item stuffs
 	public void updateBarrier() {
 		if (this.barrierActive) {
-			this.setSpriteType(DrawManager.SpriteType.ShipBarrierStatus);
+			this.setSpriteType(SpriteType.ShipBarrierStatus);
 
 			long currentTime = System.currentTimeMillis();
 
 			if (currentTime - this.barrierActivationTime >= barrier_DURATION) {
-				this.setSpriteType(DrawManager.SpriteType.Ship);
+				this.setSpriteType(SpriteType.Ship);
 				deactivatebarrier();    // deactive barrier
 			}
 		}
 	}
 
+	public void moveToEdgeLeft(boolean isLeft) {
+			if (isLeft) { this.positionX = getCurrentScreen().getWidth()-this.width; }
+	}
+	public void moveToEdgeRight(boolean isRight) {
+			if (isRight) { this.positionX = 0; }
+	}
+
+	public void detectedDoubleTap(){
+		if (Globals.getInputManager().isDoubleTab(KeyEvent.VK_RIGHT) && doubleTapCooldown.getRemainingTime() == 0 ) {
+			moveToEdgeLeft(true);
+			Globals.getLogger().info("Detected Double Tab Right");
+			doubleTapCooldown.reset();
+		}
+		if (Globals.getInputManager().isDoubleTab(KeyEvent.VK_LEFT) && doubleTapCooldown.getRemainingTime() == 0) {
+			moveToEdgeRight(true);
+			Globals.getLogger().info("Detected Double Tab Left");
+			doubleTapCooldown.reset();
+		}
+	}
```

Comment:
I add detectedDoublTap() in Ship.java. Teleport-function was executed successfully.